subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/DaemonServices.java

```diff
@@ -120,20 +120,16 @@ protected DaemonScanInfo createDaemonScanInfo(DaemonRunningStats runningStats, L
         return new DefaultDaemonScanInfo(runningStats, configuration.getIdleTimeout(), get(DaemonRegistry.class), listenerManager);
     }
 
-    protected MasterExpirationStrategy createMasterExpirationStrategy(Daemon daemon, HealthExpirationStrategy healthExpirationStrategy) {
-        return new MasterExpirationStrategy(daemon, configuration, healthExpirationStrategy);
+    protected MasterExpirationStrategy createMasterExpirationStrategy(Daemon daemon, HealthExpirationStrategy healthExpirationStrategy, ListenerManager listenerManager) {
+        return new MasterExpirationStrategy(daemon, configuration, healthExpirationStrategy, listenerManager);
     }
 
     protected HealthExpirationStrategy createHealthExpirationStrategy(DaemonMemoryStatus memoryStatus) {
         return new HealthExpirationStrategy(memoryStatus);
     }
 
-    protected DaemonHealthStats createDaemonHealthStats(DaemonRunningStats runningStats, ScheduledExecutorService scheduledExecutorService) {
-        return new DaemonHealthStats(runningStats, scheduledExecutorService);
-    }
-
-    protected ScheduledExecutorService createScheduledExecutorService() {
-        return Executors.newScheduledThreadPool(1);
+    protected DaemonHealthStats createDaemonHealthStats(DaemonRunningStats runningStats, MemoryInfo memoryInfo, ScheduledExecutorService scheduledExecutorService) {
+        return new DaemonHealthStats(runningStats, memoryInfo, scheduledExecutorService);
```

Comment:
`DaemonHealthStats` should not be using `MemoryInfo` directly, it should be receiving broadcasts.